File: TEAM A/Week 3/Jeckmu/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    termList = {}
    for t in terms:
        termList[t.split()[0]] = int(t.split()[1])

    privList = []
    for p in privacies:
        privList.append(p.split())

    t_year = int(today[:4])
    t_month = int(today[5:7])
    t_day = int(today[8:10])

    for index, p in enumerate(privList):
        year = int(p[0][:4])
        month = int(p[0][5:7])
        day = int(p[0][8:10])
        c = p[1]

        # term에 따른 month 추가
        m = termList[c]

        month += m
        while month >= 13:
            month -= 12
            year += 1

        # 비교
        if t_year > year:
            answer.append(index+1)
            continue
        elif t_year == year:
            if t_month > month:
                answer.append(index+1)
                continue
            elif t_month == month:
                if t_day >= day:
                    answer.append(index+1)
                else:
                    continue
            else:
                continue
        else:
            continue

    return answer

File: TEAM A (Season 1, 2)/Week 3/Eren/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <bits/stdc++.h>
#include <string>
using namespace std;

vector<string> split(string input, string delimeter) {
    vector<string> ret;
    long long pos = 0;
    string token = "";
    
    while((pos = input.find(delimeter)) != string::npos){
        token = input.substr(0,pos);
        ret.push_back(token);
        input.erase(0,pos + delimeter.length());
    }
    ret.push_back(input);
    return ret;
}

int atomDate(int y, int m, int d){
    return y * 28 * 12 + m * 28 + d;
}

int _term[30];

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;
    
    auto todaySplit = split(today, ".");
    int yy = stoi(todaySplit[0]);
    int mm = stoi(todaySplit[1]);
    int dd = stoi(todaySplit[2]);
    
    int todayAtom = atomDate(yy, mm, dd);
    
    for(string term: terms){
        auto t = split(term, " ");
        char name = char(t[0][0]);
        _term[name - 'A'] = stoi(t[1]);
    }
    
    int i= 1;
    for(string privacy : privacies){
        vector<string> p = split(privacy, " ");
        vector<string> date = split(p[0], ".");
        int y = stoi(date[0]);
        int m = stoi(date[1]);
        int d = stoi(date[2]);
        
        int dateAtom = atomDate(y, m, d);
        
        int term = char(p[1][0]) - 'A';
        
        if(todayAtom - dateAtom >= _term[term] * 28) 
            answer.push_back(i);
        
        // cout << _term[term] << ' ' << todayAtom - dateAtom<< endl;
        
        i++;
    }
    return answer;
}